ea.go

```diff
@@ -28,6 +37,44 @@ type ExtendedAttribute struct {
 	Flags uint8
 }
 
+// GetFileEA retrieves the extended attributes for the file represented by `handle`. The
+// `handle` must have been opened with file access flag FILE_READ_EA (0x8).
+func GetFileEA(handle windows.Handle) ([]ExtendedAttribute, error) {
+	// default buffer size to start with
+	bufLen := 1024
+	buf := make([]byte, bufLen)
+	var iosb ioStatusBlock
+	// keep increasing the buffer size until it is large enough
+	for {
+		status := getFileEA(handle, &iosb, &buf[0], uint32(bufLen), false, 0, 0, nil, true)
+		if status.Err() != nil {
+			// convert ntstatus code to windows error
+			if status.Err() == windows.ERROR_INSUFFICIENT_BUFFER || status.Err() == windows.ERROR_MORE_DATA {
+				bufLen *= 2
+				buf = make([]byte, bufLen)
+			} else {
+				return nil, fmt.Errorf("get file EA failed with: %w", status.Err())
+			}
+		} else {
```

Comment:
Don't need the else

Reply:
We need the else with break otherwise the for loop doesn't end and the code gets stuck in a loop. 
Existing code is correct.

Reply:
I should've been more clear. Keep the break just don't need the else. But we'd need something to retry the loop so we'd need a continue after `buf = make([]byte, bufLen)`, e.g. This seems mostly preference
```go
for {
    status := getFileEA(handle, &iosb, &buf[0], uint32(bufLen), false, 0, 0, nil, true)
    if status.Err() != nil {
	// convert ntstatus code to windows error
	if status.Err() == windows.ERROR_INSUFFICIENT_BUFFER || status.Err() == windows.ERROR_MORE_DATA {
		bufLen *= 2
		buf = make([]byte, bufLen)
                continue
	} else {
                return nil, fmt.Errorf("get file EA failed with: %w", status.Err())
        }
   }
   break
}



```

Reply:
I see what you mean, that does seem a bit more concise.